hoomd/md/EvaluatorPairTable.h

```diff
@@ -0,0 +1,236 @@
+                    delete[] F;
+                    });
+
+                pybind11::dict v;
+                v["V"] = pybind11::array_t<Scalar>(pybind11::array::ShapeContainer({width,}),
+                                                   pybind11::array::StridesContainer({sizeof(Scalar),}),
+                                                   V,
+                                                   free_V);
+                v["F"] = pybind11::array_t<Scalar>(pybind11::array::ShapeContainer({width,}),
+                                                   pybind11::array::StridesContainer({sizeof(Scalar),}),
+                                                   F,
+                                                   free_F);
+                v["width"] = width;
+                v["r_min"] = rmin;
+                return v;
+                }
+            #endif
+            }
+            #ifdef SINGLE_PRECISION
+            __attribute__((aligned(8)));
+            #else
+            __attribute__((aligned(16)));
+            #endif
+
+
+        //! Constructs the pair potential evaluator
+        /*! \param _rsq Squared distance between the particles
+            \param _rcutsq Squared distance at which the potential goes to 0
+            \param _params Per type pair parameters of this potential
+        */
+        DEVICE EvaluatorPairTable(Scalar _rsq, Scalar _rcutsq, const param_type& _params)
+            : rsq(_rsq), rcutsq(_rcutsq), width(_params.width), rmin(_params.rmin), V_table(_params.V_table), F_table(_params.F_table)
+            {
+            }
+
+        //! Table doesn't use diameter
+        DEVICE static bool needsDiameter() { return false; }
+        //! Accept the optional diameter values
+        /*! \param di Diameter of particle i
+            \param dj Diameter of particle j
+        */
+        DEVICE void setDiameter(Scalar di, Scalar dj) { }
+
+        //! Table doesn't use charge
+        DEVICE static bool needsCharge() { return false; }
+        //! Accept the optional diameter values
+        /*! \param qi Charge of particle i
+            \param qj Charge of particle j
+        */
+        DEVICE void setCharge(Scalar qi, Scalar qj) { }
+
+        //! Evaluate the force and energy
+        /*! \param force_divr Output parameter to write the computed force divided by r.
+            \param pair_eng Output parameter to write the computed pair energy
+            \param energy_shift If true, the potential must be shifted so that
+            V(r) is continuous at the cutoff
+            \note There is no need to check if rsq < rcutsq in this method.
+            Cutoff tests are performed in PotentialPair.
+
+            \return True if they are evaluated or false if they are not because
+            we are beyond the cutoff
+        */
+        DEVICE bool evalForceAndEnergy(Scalar& force_divr, Scalar& pair_eng, bool energy_shift)
+            {
+            // compute the force divided by r in force_divr
```

Comment:
This code is used in the inner loop, let's not trust the compiler/optimizer. Only call sqrt once for each unique value and then store that in a variable and reuse it.

Also, use fast::sqrt instead of std::sqrt.

```
Scalar rcut = fast::sqrt(rcutsq);
Scalar r = fast::sqrt(rsq);
```